sympy/stats/rv.py

```diff
@@ -1017,69 +1021,52 @@ def where(condition, given_condition=None, **kwargs):
     return pspace(condition).where(condition, **kwargs)
 
 
-def sample(expr, condition=None, size=(), **kwargs):
+def sample(expr, condition=None, size=(1,), library='scipy', numsamples=1,
+                                                                    **kwargs):
     """
     A realization of the random expression
 
+    Parameters
+    ==========
+
+    expr : Expression of random variables
+        Expression from which sample is extracted
+    condition : Expr containing RandomSymbols
+        A conditional expression
+    size : int, tuple
+        Represents size of each sample in numsamples
+    library : str
+        - 'scipy' : Sample using scipy
+        - 'numpy' : Sample using numpy
+        - 'pymc3' : Sample using PyMC3
+
+        Choose any of the available options to sample from as string,
+        by default is 'scipy'
+    numsamples : int
+        Number of samples, each with size as ``size``
+
     Examples
     ========
 
     >>> from sympy.stats import Die, sample, Normal
     >>> X, Y, Z = Die('X', 6), Die('Y', 6), Die('Z', 6)
 
-    >>> die_roll = sample(X + Y + Z) # A random realization of three dice
+    >>> die_roll = sample(X + Y + Z) # doctest: +SKIP
     >>> N = Normal('N', 3, 4)
-    >>> samp = sample(N)
-    >>> samp in N.pspace.domain.set
+    >>> samp = next(sample(N))[0] # doctest: +SKIP
```

Comment:
This appears to be a change in public API. Please go through https://github.com/sympy/sympy/wiki/Deprecating-policy and a suitable warning here.

Reply:
We have deleted the functions `sample_iter_lambdify` and `sample_iter_subs`. Also, in `sample` and `sample_iter`, there is a change in the return type. So I think I should add warnings when someone tries to access `sample_iter_lambdify` and `sample_iter_subs`. Rest is only the return type change?

Reply:
> We have deleted the functions `sample_iter_lambdify` and `sample_iter_subs`

Why? Were these functions available in `sympy.stats` namespace?
@Upabjojr Any thoughts on removing these two functions?

> Rest is only the return type change?

May be, we should warning for these too as the output of public functions are changed.

Reply:
My opinion matches with the following from https://github.com/sympy/sympy/issues/19061#issuecomment-608267737

> However, I think that this doesn't need much deprecation if it works just fine, but there don't have to be any addition for it.

We can keep both the functions and warnings can be added for the change returned types everywhere.

Reply:
See my suggestions below.

Reply:
> Why? Were these functions available in `sympy.stats` namespace?

No, there were not present in `sympy.stats` namespace.
Also, if they are not used anywhere, and `sample_iter`, now is capable of handling the need using `lambdify`( which was previously `sample_iter_lambdify`) and `sample_iter_subs` is too slow and doesn't follow the discussions in #14261. So, IMHO we should remove these two functions if not now then in further versions.